File: 프로그래머스/2/84512. 모음 사전/모음 사전.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

int solution(string word) {
    string s = "AEIOU";
    int arr[] = { 781, 156, 31, 6, 1 }, answer = 0;
    
    for(int i = 0; i < word.size(); i++)
        answer += s.find(word[i]) * arr[i] + 1; 
    
    return answer;
}

File: 프로그래머스/2/84512. 모음 사전/모음 사전.java
class Solution {
    public int solution(String word) {
        int[] arr = new int[5];
        arr[4] = 1;
       
        String s = "AEIOU";
        int x = 5, answer = 0;
        
        for(int i = 3; i >= 0; i--)
            arr[i] = arr[i + 1]  * 5 + 1;
        
        for(int i = 0 ; i < word.length(); i++)
            answer += s.indexOf(word.charAt(i)) * arr[i] + 1;    
        
        return answer;
    }
}